src/Services/RequestHelpers.ts

```diff
@@ -29,6 +30,16 @@ export class RequestHelpers {
     return ['User-Agent', `AuditJS/${pack.version} (${environment} ${environmentVersion}; ${system})`];
   }
 
+  public static getAgent(insecure = false): Agent | undefined {
+    if (insecure) {
+      return new HttpsAgent({
```

Comment:
In the case of insecure, the proxy setting is ignored...

Reply:
> In the case of insecure, the proxy setting is ignored...

I think I see what you mean. I added some tests of the `getAgent()` method in: 33166519132b054e4f070f28d77e4b2fdf9d614e, but I'm unsure how best to deal with the case of an 'insecure proxy`. What would we want to see it do? (What to [expect()](https://github.com/sonatype-nexus-community/auditjs/blob/master/src/Services/RequestHelpers.spec.ts#L68) in this test?)

I'm unsure how the insecure url and proxy would interact.

A PR to handle this case would be great!

And/Or a new Issue to track this case.

Reply:
I think you assumptions are correct. This is what would be a combination of proxy and insecure. 
To achieve this, I think we should pass the insecure flag to https-proxy-agent. It can take either a string (like now) or an option object. I think this object can have this flags.